components/jaggery-core/org.jaggeryjs.jaggery.core/src/main/java/org/jaggeryjs/jaggery/core/manager/JaggeryDeployerManager.java

```diff
@@ -0,0 +1,640 @@
+                CommonManager.setJaggeryContext(sharedContext);
+                RhinoEngine engine = sharedContext.getEngine();
+                org.mozilla.javascript.Context cx = engine.enterContext();
+                ServletContext servletContext = (ServletContext) sharedContext
+                        .getProperty(org.jaggeryjs.hostobjects.web.Constants.SERVLET_CONTEXT);
+                ScriptableObject sharedScope = sharedContext.getScope();
+
+                Object[] scripts = arr.toArray();
+                for (Object script : scripts) {
+                    if (!(script instanceof String)) {
+                        log.error("Invalid value for initScripts/destroyScripts in jaggery.conf : " + script);
+                        continue;
+                    }
+                    String path = (String) script;
+                    path = path.startsWith("/") ? path : "/" + path;
+                    Stack<String> callstack = CommonManager.getCallstack(sharedContext);
+                    callstack.push(path);
+                    String[] parts = WebAppManager.getKeys(servletContext.getContextPath(), path, path);
+                    ScriptCachingContext sctx = new ScriptCachingContext(sharedContext.getTenantDomain(), parts[0],
+                            parts[1], parts[2]);
+                    sctx.setSecurityDomain(new JaggerySecurityDomain(path, servletContext));
+                    engine.exec(new ScriptReader(servletContext.getResourceAsStream(path)) {
+                        @Override protected void build() throws IOException {
+                            try {
+                                sourceReader = new StringReader(HostObjectUtil.streamToString(sourceIn));
+                            } catch (ScriptException e) {
+                                // throw new IOException(e);
+                            }
+                        }
+                    }, sharedScope, sctx);
+                }
+            } catch (ScriptException e) {
+                log.error(e.getMessage(), e);
+            } finally {
+                if (org.mozilla.javascript.Context.getCurrentContext() != null) {
+                    RhinoEngine.exitContext();
+                }
+            }
+        }
+    }
+
+    private static void setDisplayName(Context context, JSONObject obj) {
+        if (obj == null) {
+            return;
+        }
+        String dName = (String) obj.get(JaggeryCoreConstants.JaggeryConfigParams.DISPLAY_NAME);
+        if (dName != null) {
+            context.setDisplayName(dName);
+        }
+    }
+
+    private static void addUrlMappings(Context context, JSONObject obj) {
+        Object test = context.getServletContext().getAttribute("org.jaggeryjs.serveFunction");
+        JSONArray arr = null;
+        if (test != null) {
+            // URL mapping for progamticaly
+            arr = new JSONArray();
+            JSONObject js1 = new JSONObject();
```

Comment:
Add proper variable names, don't use js1, js2 sort of names, and it doesn't explain what actually it's doing.